CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -288,6 +314,75 @@ - (BOOL)startWithTaskGroup:(dispatch_group_t)taskGroup error:(NSError *__autorel
     return YES;
 }
 
+/**
+ Builds a TDReplicator object from a CDTPush/PullReplication configuration.
+ */
+- (TDReplicator *)buildTDReplicatorFromConfiguration:(CDTAbstractReplication *)configuration
```

Comment:
It's confusing to have `configuration` as an argument to this method because it's already a member of this class, so there is no need to pass the argument to this method.

You have two choices - either make it a static method which takes the argument, or keep it as an instance method and remove the argument.